src/typeInference/annotateAST.ts

```diff
@@ -379,4 +372,17 @@ const annotationFunctions: {
 		}
 		return type;
 	},
+	flworExpr: (ast: IAST, context: AnnotationContext): SequenceType => {
+		return annotateFlworExpression(ast, context, annotate);
+	},
+	varRef: (ast: IAST, context: AnnotationContext): SequenceType => {
+		return annotateVarRef(ast[1] as IAST, context);
+	},
+	returnClause: (ast: IAST, context: AnnotationContext): SequenceType => {
```

Comment:
This is not really an AST node, Its more a property of the FlworExpr so I think it fits better inside annotateFlworExpression.